staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -337,6 +352,22 @@ func (e *Store) ListPredicate(ctx context.Context, p storage.SelectionPredicate,
 
 // Create inserts a new item according to the unique key from the object.
 func (e *Store) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
```

Comment:
```go
func cleanNothing(ctx, bool){}

Create() {
  var cleanupFn cleanupFunc = cleanNothing
  success := false
  defer func(){
    cleanup(ctx, success)
  }

  if e.BeginCreate != nil{
    fn, err := e.BeginCreate
    if err != nil{
    }
    cleanFn = fn
  }

  // where the existing duplicate cleanup block is
  success = true
```

Reply:
Given the ordering we talked about, I can do something similar but not quite this.  I'm going to add it as a distinct commit.  If you like it, we keep it.  If not, we abandon just that.